src/otx/core/data/module.py

```diff
@@ -217,6 +275,63 @@ def predict_dataloader(self) -> DataLoader:
             persistent_workers=config.num_workers > 0,
         )
 
+    def unlabeled_dataloader(self) -> dict[str, DataLoader] | None:
+        """Returns a dictionary of unlabeled dataloaders.
+
+        The method creates and returns dataloaders for unlabeled datasets based on the configuration settings.
+        If the data root is not specified in the configuration, it returns None.
+
+        Returns:
+            dict[str, DataLoader] | None: A dictionary containing unlabeled dataloaders, where the keys are the names of
+            the datasets and the values are the corresponding DataLoader objects.
+        """
+        config = self.config.unlabeled_subset
+        if self.config.unlabeled_subset.data_root is None:
+            return None
+
+        common_args = {
+            "batch_size": config.batch_size,
+            "num_workers": config.num_workers,
+            "pin_memory": True,
+            "persistent_workers": config.num_workers > 0,
+        }
+
+        dataloaders = {}
+        if isinstance(config.transforms, dict):
+            log.warning(f"Unlabeled dataset has multiple transforms : {list(config.transforms.keys())}")
+            common_args["worker_init_fn"] = lambda _: torch.manual_seed(0)  # type: ignore[assignment]
+            for key in config.transforms:
+                dataset = self._get_dataset(key)
+                # For unlabeled datasets using Multi-Transforms, must use generators and samplers
+                # with the same seed to get the same data.
+                generator = torch.Generator().manual_seed(0)
```

Comment:
same about seed here

Reply:
This is to enable multiple unlabeled dataloaders to sample the same batch of data.